pkg/resources/spark/service.go

```diff
@@ -24,8 +24,9 @@ func NewMasterService(sc *dcv1alpha1.SparkCluster) *corev1.Service {
 	}
 	if util.BoolPtrIsTrue(sc.Spec.EnableDashboard) {
 		ports = append(ports, corev1.ServicePort{
-			Name:     "tcp", // named tcp to prevent istio from sniffing for Host
-			Port:     sc.Spec.DashboardPort,
+			// deliberately named tcp to prevent istio from sniffing for Host
+			Name:     "tcp",
```

Comment:
I believe you can _prefix_ as well (i.e. `tcp-dashboard`) instead of just calling it "tcp"